libs/domain/product/src/mixins/product.mixin.ts

```diff
@@ -34,15 +30,23 @@ export const ProductMixin = <
   class ProductMixinClass extends superClass {
     @signalProperty({ reflect: true }) sku?: string;
 
-    protected productController = new ProductController(this);
+    protected productService = resolve(ProductService, null);
+    protected contextController = new ContextController(this);
+
+    protected $context = signal(
+      this.contextController.get<string>(ProductContext.SKU)
+    );
 
-    @asyncState()
-    protected product = valueType(this.productController.getProduct());
+    protected setProductContext = effect(() => {
+      if (this.sku && !this.$context()) {
```

Comment:
I'm wondering if we should check for `!this.$context()` as my reasoning is, that if we pass `sku` as a property, we should override context (from parent element)